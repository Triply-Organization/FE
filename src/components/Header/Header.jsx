import { UserOutlined } from '@ant-design/icons';
import { AreaChartOutlined } from '@ant-design/icons';
import { Button, Dropdown, Menu } from 'antd';
import { Avatar } from 'antd';
import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { AiOutlineHome } from 'react-icons/ai';
import { AiOutlineLogout } from 'react-icons/ai';
import { TbTicket } from 'react-icons/tb';
import { Link, useNavigate } from 'react-router-dom';

import logo from '../../assets/images/logo.png';
import CurrencySelect from '../CurrencySelect/CurrencySelect';
import LanguageSelect from './../LanguageSelect/LanguageSelect';
import './Header.scss';
import MobileNav from './MobileNav/MobileNav';
import Navbar from './Navbar';

export default function Header() {
  // state set for active tab
  const [activeTab, setActiveTab] = useState(() => {
    if (location.pathname === '/') {
      return 0;
    } else {
      return 1;
    }
  });

  const token = localStorage.getItem('token');
  const roles = JSON.parse(localStorage.getItem('user'))?.roles || [];
  const id = JSON.parse(localStorage.getItem('user'))?.id || 1;

  // state set for window srollY
  const [scrollY, setScrollY] = useState(window.scrollY);
  // state set for window width
  const [width, setWidth] = useState(window.innerWidth);
  // state set for mobileNav status
  const [mobileNavStatus, setMobileNavStatus] = useState(false);
  const { t } = useTranslation();
  const navigate = useNavigate();

  const handleLogout = () => {
    localStorage.removeItem('token');
    localStorage.removeItem('user');
  };

  const userDataLogin = (
    <Menu
      items={[
        {
          key: '1',
          type: 'group',
          label: `Hi ${
            JSON.parse(localStorage.getItem('user'))?.name || 'there'
          }`,
          children: [
            roles.includes('ROLE_USER')
              ? {
                  key: '1-1',
                  label: (
                    <span className="header__language-item">
                      <Link
                        to={`/setting-account/${id}`}
                        className="header__language-words"
                      >
                        {t('header.logged_in.my_profile')}
                      </Link>
                    </span>
                  ),
                }
              : null,
            roles.includes('ROLE_USER')
              ? {
                  key: '1-2',
                  label: (
                    <span className="header__language-item">
                      <Link to="/my-tours" className="header__language-words">
                        {t('header.logged_in.tour')}
                      </Link>
                    </span>
                  ),
                }
              : null,
            roles.includes('ROLE_ADMIN')
              ? !roles.includes('ROLE_USER')
                ? {
                    key: '1-21',
                    label: (
                      <span className="header__language-item">
                        <Link to="/admin" className="header__language-words">
                          {t('header.logged_in.dashboard')}
                        </Link>
                      </span>
                    ),
                  }
                : null
              : !roles.includes('ROLE_USER')
              ? {
                  key: '1-21',
                  label: (
                    <span className="header__language-item">
                      <Link to="/cms" className="header__language-words">
                        {t('header.logged_in.dashboard')}
                      </Link>
                    </span>
                  ),
                }
              : null,
            {
              key: '1-3',
              label: (
                <span className="header__language-item">
                  <Link
                    onClick={handleLogout}
                    to="/login"
                    className="header__language-words"
                  >
                    {t('cta.logout')}
                  </Link>
                </span>
              ),
            },
          ],
        },
      ]}
    />
  );

  //---------- User Data When Login ----------->
  const userDataLoginMobile = [
    roles.includes('ROLE_USER')
      ? {
          title: `${t('header.logged_in.my_profile')}`,
          icon: <UserOutlined />,
          link: `/setting-account/${id}`,
        }
      : null,
    roles.includes('ROLE_USER')
      ? {
          title: `${t('header.logged_in.tour')}`,
          icon: <TbTicket />,
          link: '/my-tours',
        }
      : null,
    roles.includes('ROLE_ADMIN')
      ? !roles.includes('ROLE_USER')
        ? {
            title: `${t('header.logged_in.dashboard')}`,
            icon: <AreaChartOutlined />,
            link: '/admin',
          }
        : null
      : !roles.includes('ROLE_USER')
      ? {
          title: `${t('header.logged_in.dashboard')}`,
          icon: <AreaChartOutlined />,
          link: '/cms',
        }
      : null,

    {
      title: `${t('cta.logout')}`,
      icon: <AiOutlineLogout />,
      link: '/login',
      onClick: handleLogout,
    },
  ];

  //--------------- Nav Item ----------------->
  const navItem = [
    {
      title: `${t('header.home')}`,
      icon: <AiOutlineHome />,
      subnav: [],
      link: '/',
    },
    {
      title: `${t('header.tour')}`,
      icon: <TbTicket />,
      subnav: [],
      link: '/tours',
    },
  ];

  //------------- User Not Login ------------>
  const userDataNoLogin = (
    <Menu
      items={[
        {
          disabled: true,
          key: '1',
          label: (
            <span className="header__language-heading header__language-item">
              <span className="header__language-abbre">
                {t('header.not_logged_in.my_account')}
              </span>
            </span>
          ),
        },
        {
          key: '2',
          label: (
            <Button type="primary" size="large">
              <Link to="/login">{t('cta.login')}</Link>
            </Button>
          ),
        },
        {
          key: '3',
          label: (
            <Button type="default" size="large">
              <Link to="/register">{t('cta.register')}</Link>
            </Button>
          ),
        },
      ]}
    />
  );

  const handleSetActiveTab = id => {
    setActiveTab(id);
  };

  const onChangeNavbarStatus = () => {
    setMobileNavStatus(!mobileNavStatus);
  };

  useEffect(() => {
    window.addEventListener('scroll', () => {
      setScrollY(window.scrollY);
    });
  }, [scrollY]);

  useEffect(() => {
    window.addEventListener('resize', () => {
      setWidth(window.innerWidth);
    });
  }, [width]);

  return (
    <header
      className={
        scrollY > 0
          ? 'header__wrapper header__wrapper--transition'
          : 'header__wrapper'
      }
    >
      <div className="header__left-side">
        <div className="header__logo-wrapper" onClick={() => navigate('/')}>
          <img src={logo} alt="logo" className="header__logo" />
        </div>
        {width < 1023 ? null : (
          <Navbar
            activeTab={activeTab}
            handleSetActiveTab={handleSetActiveTab}
            navItem={navItem}
          />
        )}
      </div>
      <div className="header__right-side">
        {width < 1023 ? (
          <MobileNav
            mobileNavStatus={mobileNavStatus}
            onChangeNavbarStatus={onChangeNavbarStatus}
            navItem={navItem}
            userDataLoginMobile={userDataLoginMobile}
          />
        ) : (
          <>
            <div className="header__multi-currency">
              <CurrencySelect />
            </div>
            <div className="header__multi-lang-wrapper">
              <LanguageSelect />
            </div>

            {token ? (
              <div className="header__account-wrapper">
                <Dropdown overlay={userDataLogin} placement="bottomRight" arrow>
                  <div className="header__account-icon">
                    <Avatar icon={<UserOutlined />} />
                  </div>
                </Dropdown>
              </div>
            ) : (
              <div className="header__account-wrapper">
                <Dropdown
                  overlay={userDataNoLogin}
                  overlayClassName="header__account-dropdown"
                  placement="bottomRight"
                  arrow
                >
                  <div className="header__account-icon">
                    <Avatar
                      src={
                        JSON.parse(localStorage.getItem('user'))?.avatar || null
                      }
                      icon={
                        JSON.parse(localStorage.getItem('user'))
                          ?.avatar ? null : (
                          <UserOutlined />
                        )
                      }
                    />
                  </div>
                </Dropdown>
              </div>
            )}
          </>
        )}
      </div>
    </header>
  );
}
