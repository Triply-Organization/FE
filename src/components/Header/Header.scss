.header__wrapper {
  position: fixed;
  top: 0;
  background-color: #fff;
  width: 100%;
  display: flex;
  align-items: center;
  height: 90px;
  padding: 0 30px;
  transition: all 0.2s ease;
  z-index: 999;
  box-shadow: 0 1px 8px 2px rgba(0, 0, 0, 0.3);

  &.header__wrapper--transition {
    height: 74px;
    .header__logo-wrapper {
      transform: scale(0.8);
      transform-origin: left;
    }
  }
}

.header__left-side {
  display: flex;
  flex: 1;
  height: 100%;
  align-items: center;
}

.header__logo-wrapper {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 4%;
  padding: 25px 0;
  transition: all 0.2s ease;
  cursor: pointer;
}

.header__logo {
}

.header__right-side {
  display: flex;
  height: 100%;
  align-items: center;
}

.header__account-icon {
  & > .ant-avatar.ant-avatar-icon {
    font-size: 18px;
    line-height: 18px;
    color: #000;
    background: #fff;
    border: 1px solid #ccc;
    padding: 18px;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    box-shadow: 0 2px 0 rgb(0 0 0 / 2%);

    & > span.anticon.anticon-user {
      border: 1px solid black;
      border-radius: 50%;
      padding: 2px;
    }
  }
}

.header__language-words {
  color: #000;
  display: block;

  &:hover {
    color: #dc834e;
  }
}

.header__multi-currency {
  margin-right: 12px;
}

button.ant-btn.ant-btn-default.ant-dropdown-trigger {
  border-radius: 30px;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1rem;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 18px;

  &:hover {
    border: 1px solid #d9d9d9;
  }
}

li.ant-dropdown-menu-item.ant-dropdown-menu-item-disabled.ant-dropdown-menu-item-only-child {
  font-size: 1.1rem;
  color: black;
  font-weight: 600;
  cursor: default;
  margin-bottom: 4px;
}

li.ant-dropdown-menu-item.ant-dropdown-menu-item-only-child {
  font-size: 1rem;

  &:hover {
    color: #dc834e;
  }
}

.header__multi-lang-wrapper {
  margin-right: 24px;
}

button.ant-btn.ant-btn-default.ant-dropdown-trigger.header__multi-lang-btn {
  display: flex;
  align-items: center;
  justify-content: center;
}

.header__multi-lang-icon {
  margin-right: 8px;
  font-size: 1.2rem;
}

.header__multi-lang-words {
  font-size: 1rem;
}

ul.ant-dropdown-menu.ant-dropdown-menu-root.ant-dropdown-menu-vertical.ant-dropdown-menu-light {
  padding: 10px;
  border-radius: 10px;
}

.header__account-dropdown {
  &
    ul.ant-dropdown-menu.ant-dropdown-menu-root.ant-dropdown-menu-vertical.ant-dropdown-menu-light {
    min-width: 250px;
  }

  li {
    &:hover {
      background: none;
    }
  }

  & button {
    width: 100%;
  }
}

.ant-btn.ant-btn-default.ant-dropdown-trigger {
  &.header__multi-currency-container {
    &:hover {
      color: #000;
    }
  }

  &.header__multi-lang-btn {
    &:hover {
      color: #000;
    }
  }
}

.header__account-wrapper {
  position: relative;
  &::before {
    content: '';
    position: absolute;
    height: 65%;
    color: #000;
    left: -11px;
    top: 50%;
    transform: translateY(-50%);
    width: 1px;
    border-left: 1px solid #00000030;
  }
}

.header-navbar__wrapper {
  flex: 1;
}
