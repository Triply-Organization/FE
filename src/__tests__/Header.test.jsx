import {
  act,
  fireEvent,
  getByText,
  render,
  screen,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import { describe, expect, test } from 'vitest';

import Header from '../components/Header/';

describe('Header component', () => {
  test('should render correctly', () => {
    const result = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(result).toMatchSnapshot();
  });

  test('should correct header class name', () => {
    const ctn = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );

    const header = ctn.queryByTestId('header');

    expect(header).toBeInTheDocument();

    expect(header.className).toBe('header__wrapper');

    fireEvent.scroll(window, { target: { scrollY: 100 } });

    expect(header.className).toBe(
      'header__wrapper header__wrapper--transition',
    );
  });

  test('should correct activeTab', () => {
    const ctn = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
  });
});
