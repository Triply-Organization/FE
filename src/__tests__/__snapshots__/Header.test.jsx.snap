// Vitest Snapshot v1

exports[`Header component > should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header__wrapper"
      >
        <div
          class="header__left-side"
        >
          <div
            class="header__logo-wrapper"
          >
            <img
              alt="logo"
              class="header__logo"
              src="/src/assets/images/logo.png"
            />
          </div>
          <nav
            class="header__navbar-wrapper"
            data-testid="header-navbar"
          >
            <ul
              class="header__navbar"
            >
              <li
                class="header__navbar-item"
                data-testid="header-navbar-item"
              >
                <a
                  class="header__navbar-link header__navbar-link--active"
                  href="/"
                >
                  header.home
                </a>
              </li>
              <li
                class="header__navbar-item"
                data-testid="header-navbar-item"
              >
                <a
                  class="header__navbar-link"
                  href="/tours"
                >
                  header.tour
                </a>
              </li>
            </ul>
          </nav>
        </div>
        <div
          class="header__right-side"
        >
          <div
            class="header__multi-currency"
          >
            <div>
              <div
                class="ant-select ant-select-single ant-select-show-arrow"
                style="width: 80px;"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-search"
                  >
                    <input
                      aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                      aria-autocomplete="list"
                      aria-controls="rc_select_TEST_OR_SSR_list"
                      aria-haspopup="listbox"
                      aria-owns="rc_select_TEST_OR_SSR_list"
                      autocomplete="off"
                      class="ant-select-selection-search-input"
                      id="rc_select_TEST_OR_SSR"
                      readonly=""
                      role="combobox"
                      style="opacity: 0;"
                      type="search"
                      unselectable="on"
                      value=""
                    />
                  </span>
                  <span
                    class="ant-select-selection-placeholder"
                  />
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="header__multi-lang-wrapper"
          >
            <div
              class="ant-select ant-select-single ant-select-show-arrow"
              style="width: 150px;"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                    aria-autocomplete="list"
                    aria-controls="rc_select_TEST_OR_SSR_list"
                    aria-haspopup="listbox"
                    aria-owns="rc_select_TEST_OR_SSR_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="rc_select_TEST_OR_SSR"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-placeholder"
                />
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
          <div
            class="header__account-wrapper"
          >
            <div
              class="ant-dropdown-trigger header__account-icon"
            >
              <span
                class="ant-avatar ant-avatar-circle ant-avatar-icon"
              >
                <span
                  aria-label="user"
                  class="anticon anticon-user"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="user"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header__wrapper"
    >
      <div
        class="header__left-side"
      >
        <div
          class="header__logo-wrapper"
        >
          <img
            alt="logo"
            class="header__logo"
            src="/src/assets/images/logo.png"
          />
        </div>
        <nav
          class="header__navbar-wrapper"
          data-testid="header-navbar"
        >
          <ul
            class="header__navbar"
          >
            <li
              class="header__navbar-item"
              data-testid="header-navbar-item"
            >
              <a
                class="header__navbar-link header__navbar-link--active"
                href="/"
              >
                header.home
              </a>
            </li>
            <li
              class="header__navbar-item"
              data-testid="header-navbar-item"
            >
              <a
                class="header__navbar-link"
                href="/tours"
              >
                header.tour
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        class="header__right-side"
      >
        <div
          class="header__multi-currency"
        >
          <div>
            <div
              class="ant-select ant-select-single ant-select-show-arrow"
              style="width: 80px;"
            >
              <div
                class="ant-select-selector"
              >
                <span
                  class="ant-select-selection-search"
                >
                  <input
                    aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                    aria-autocomplete="list"
                    aria-controls="rc_select_TEST_OR_SSR_list"
                    aria-haspopup="listbox"
                    aria-owns="rc_select_TEST_OR_SSR_list"
                    autocomplete="off"
                    class="ant-select-selection-search-input"
                    id="rc_select_TEST_OR_SSR"
                    readonly=""
                    role="combobox"
                    style="opacity: 0;"
                    type="search"
                    unselectable="on"
                    value=""
                  />
                </span>
                <span
                  class="ant-select-selection-placeholder"
                />
              </div>
              <span
                aria-hidden="true"
                class="ant-select-arrow"
                style="user-select: none;"
                unselectable="on"
              >
                <span
                  aria-label="down"
                  class="anticon anticon-down ant-select-suffix"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="down"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          class="header__multi-lang-wrapper"
        >
          <div
            class="ant-select ant-select-single ant-select-show-arrow"
            style="width: 150px;"
          >
            <div
              class="ant-select-selector"
            >
              <span
                class="ant-select-selection-search"
              >
                <input
                  aria-activedescendant="rc_select_TEST_OR_SSR_list_0"
                  aria-autocomplete="list"
                  aria-controls="rc_select_TEST_OR_SSR_list"
                  aria-haspopup="listbox"
                  aria-owns="rc_select_TEST_OR_SSR_list"
                  autocomplete="off"
                  class="ant-select-selection-search-input"
                  id="rc_select_TEST_OR_SSR"
                  readonly=""
                  role="combobox"
                  style="opacity: 0;"
                  type="search"
                  unselectable="on"
                  value=""
                />
              </span>
              <span
                class="ant-select-selection-placeholder"
              />
            </div>
            <span
              aria-hidden="true"
              class="ant-select-arrow"
              style="user-select: none;"
              unselectable="on"
            >
              <span
                aria-label="down"
                class="anticon anticon-down"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="down"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
        <div
          class="header__account-wrapper"
        >
          <div
            class="ant-dropdown-trigger header__account-icon"
          >
            <span
              class="ant-avatar ant-avatar-circle ant-avatar-icon"
            >
              <span
                aria-label="user"
                class="anticon anticon-user"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="user"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </span>
            </span>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
